### PR TITLE
feat(core): support default slot for SlotMixin

### PR DESCRIPTION
**Pull request by jorenbroekema**

### CODE REVIEW

File: packages/core/src/SlotMixin.js [hunk 1]
```diff
@@ -49,7 +49,13 @@ const SlotMixinImplementation = superclass =>
     _connectSlotMixin() {
       if (!this.__isConnectedSlotMixin) {
         Object.keys(this.slots).forEach(slotName => {
-          if (!Array.from(this.children).find(el => el.slot === slotName)) {
+          // default slot means, we just check if there's child elements, otherwise check for slottable by slot-name
+          const hasSlottableFromUser =
+            slotName === ''
+              ? this.children.length > 0
```
Comment (tlouisse) [inline packages/core/src/SlotMixin.js:55]: If there's a different named slot already it would unrightfully pass I thunk

Comment (jorenbroekema) [reply packages/core/src/SlotMixin.js:55]: Ohh true

Comment (jorenbroekema) [reply packages/core/src/SlotMixin.js]: Amended, added a test as well which failed before the update due to the edge case you pointed out 👍

File: packages/core/src/SlotMixin.js [hunk 2]
```diff
@@ -49,7 +49,15 @@ const SlotMixinImplementation = superclass =>
     _connectSlotMixin() {
       if (!this.__isConnectedSlotMixin) {
         Object.keys(this.slots).forEach(slotName => {
-          if (!Array.from(this.children).find(el => el.slot === slotName)) {
+          const hasSlottableFromUser =
+            slotName === ''
+              ? // for default slot (''), we can't use el.slot because polyfill for IE11
+                // will do .querySelector('[slot=]') which produces a fatal error
+                // therefore we check if there's children that do not have a slot attr
+                Array.from(this.children).find(el => !el.hasAttribute('slot'))
```
Comment (tlouisse) [inline packages/core/src/SlotMixin.js:57]: Following that reasoning, you should also check via getAttribute in the line below? :P

Comment (tlouisse) [reply packages/core/src/SlotMixin.js:57]: Ah nvm, didn't read it correctly

**Review by tlouisse — APPROVED** [review]

**Comment by lpoulter** [comment]
I was experiencing this issue, I tested this change locally and it works as expected for me.
